adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/VertxBasedMqttProtocolAdapter.java

```diff
@@ -201,18 +201,16 @@ private void handleEndpointConnection(final MqttEndpoint endpoint) {
                     MqttUsernamePassword authObject = MqttUsernamePassword.create(endpoint,
```

Comment:
Maybe the `MqttUsernamePassword` could be more explicit in the API not accepting the MqttEndpoint but username and password directly and maybe calling it `MqttAuth` or something like that ?